main/src/tutorial/tutorial.flix

```diff
@@ -0,0 +1,237 @@
+        case _                                      => Constant.Top
+    }
+
+    //
+    // ... and the greatest lower bound:
+    //
+    def glb(e1: Constant, e2: Constant): Constant = match (e1, e2) with {
+        case (Constant.Top, x)                      => x
+        case (x, Constant.Top)                      => x
+        case (Constant.Cst(n1), Constant.Cst(n2))   => if (n1 == n2) e1 else Constant.Bot
+        case _                                      => Constant.Bot
+    }
+
+    //
+    // This complete the specification of the lattice. To associate these
+    // functions with the type Constant we have to write some magic:
+    //
+    let Constant<> = (Constant.Bot, Constant.Top, leq, lub, glb)
+    // NB: In the future, this wierd syntax will be replaced by type classes.
+
+    //
+    // Finally it is handly to have a small function to lift an integer into
+    // an element of the constant propagation lattice. Strictly speaking,
+    // this is not necessary.
+    //
+    def alpha(i: Int): Constant = Constant.Cst(i)
+
+    //
+    // We can now define the semantic rules of the analysis!
+    //
+
+    //
+    // The first rule models literal statements r = c and lifts the constant
+    // into the constant propagation lattice.
+    //
+    LocalVar(r, alpha(c)) :- LitStm(r, c).
+
+    //
+    // The second rule is more interesting. This says that if we have an
+    // addition statement r = x + y, and the value of x is v1 and of y is v2
+    // then the value of r is at least the value of the sum of v1 and v2.
+    //
+    // Here sum is a reference to a transfer function implemented below.
+    //
+    LocalVar(r, sum(v1, v2)) :- AddStm(r, x, y),
+                                LocalVar(x, v1),
+                                LocalVar(y, v2).
+
+    //
+    // The third rule is similar, but for division:
+    //
+    LocalVar(r, div(v1, v2)) :- DivStm(r, x, y),
+                                LocalVar(x, v1),
+                                LocalVar(y, v2).
+
+    //
+    // We must define the sum transfer function:
+    //
+    def sum(e1: Constant, e2: Constant): Constant = match (e1, e2) with {
+        case (Constant.Bot, _)                      => Constant.Bot
+        case (_, Constant.Bot)                      => Constant.Bot
+        case (Constant.Cst(n1), Constant.Cst(n2))   => Constant.Cst(n1 + n2)
+        case _                                      => Constant.Top
+    }
+
+    //
+    // And the div transfer function:
+    //
+    def div(e1: Constant, e2: Constant): Constant = match (e1, e2) with {
+        case (_, Constant.Bot) => Constant.Bot
+        case (Constant.Bot, _) => Constant.Bot
+        case (Constant.Cst(n1), Constant.Cst(n2)) => if (n2 == 0) Constant.Bot else Constant.Cst(n1 / n2)
+        case _ => Constant.Top
+    }
+
+    // !! Notice the special case to avoid division-by-zero in the analysis itself!
```

Comment:
Nitpick: I would probably format this as `!! Notice the special case ... analysis itself !!`. I think the symmetric exclamation marks makes it look nicer.